opensrp-app/src/main/java/org/smartregister/Context.java

```diff
@@ -580,12 +591,46 @@ public AllSettings allSettings() {
 
     public AllSharedPreferences allSharedPreferences() {
         if (allSharedPreferences == null) {
-            allSharedPreferences = new AllSharedPreferences(
-                    getDefaultSharedPreferences(this.applicationContext));
+            allSharedPreferences = new AllSharedPreferences(createSharedPreferences(this.applicationContext));
         }
         return allSharedPreferences;
     }
 
+    private SharedPreferences createSharedPreferences(android.content.Context context) {
+        SyncConfiguration syncConfiguration = CoreLibrary.getInstance().getSyncConfiguration();
+        
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
+                && syncConfiguration != null
+                && syncConfiguration.encryptSharedPreferences()) {
+
+            return createEncryptedSharedPreferences(context);
+        } else {
+            return getDefaultSharedPreferences(context);
+        }
+    }
+
+    private SharedPreferences createEncryptedSharedPreferences(android.content.Context context) {
+        SharedPreferences sharedPreferences = null;
+
+        try {
+            String masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC);
+
+            sharedPreferences = EncryptedSharedPreferences.create(
+                    SHARED_PREFERENCES_FILENAME,
+                    masterKeyAlias,
+                    context,
+                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+            );
+        } catch (GeneralSecurityException e) {
+            Timber.e(e);
+        } catch (IOException e) {
+            Timber.e(e);
+        }
```

Comment:
What is the difference in these?In case this fails what is the fall back, stop app or use normal preferences?

Reply:
Updated to fall back to unencrypted SharedPreferences on error